### DISCUSSION TITLE
How can I specify the edition for Rust?

### DISCUSSION THREAD

**Discussion by dclong** [evcxr/evcxr#200] [Q&A]
I have Rust 1.56.0, however, some of the Rust 2021 edition features is not enabled when I run code in Jupyter/Lab notebook. I wonder how I can check/set the edition of Rust when using evcxr_jupyter?

**Comment by bjorn3** [comment]
Looks like it is hard-coded to the 2018 edition currently: https://github.com/google/evcxr/blob/d83d1a49297e4807f0804109117b7c9a55c62c3b/evcxr/src/module.rs#L232

**Comment by davidlattimore** [comment]
Thanks for bringing that up. I've just pushed updates to github that increase the minimum rust version to 1.56 and switches to the 2021 edition. I won't release to crates.io just yet, but you can try installing it with the following command:

```sh
cargo install --force --git https://github.com/google/evcxr.git evcxr_jupyter
```

**Reply by dclong** [reply]
The command `cargo install --force --git https://github.com/google/evcxr.git --branch main evcxr_jupyter` fails to install evcxr_jupyter.

**Reply by davidlattimore** [reply]
Hmm, you're right. So it does. I guess we need to update the instructions for how to install to git. I couldn't find a way to get cargo install to install it directly from git.

Until we figure it out, you can install by cloning the evcxr repository then installing from that. e.g.

```sh
git clone https://github.com/google/evcxr.git
cargo install --path evcxr/evcxr_jupyter
```
